andela/mrm_push
# CON-137 Auto-refresh push notifications

#### Description
This Pull request enables auto-refreshing of push notification channels using celery and celery beat.

#### Type of Change
Added Celery to the repo. This is used in scheduling the push notification instead of hitting the refresh endpoint manually. 
#### How has this been tested


#### How should this be manually tested?
1. Clone the repo: `git clone https://github.com/andela/mrm_push.git`
2. Setup project according to Readme.md
3. checkout to branch `story/CON-137-auto-refresh-push-notifications`
4. Open two terminal windows on the root folder of the project and run the following commands:
       a. `celery worker -A cworker.celery --loglevel=info` on the first terminal.
       b. `celery beat -A cworker.celery --schedule=/tmp/celerybeat-schedule --loglevel=INFO --pidfile=/tmp/celerybeat.pid` on the second terminal window.
5. In the first terminal you will see the following:
<img width="1139" alt="Screenshot 2019-07-23 at 10 24 13" src="https://user-images.githubusercontent.com/23398223/61691394-201a2600-ad34-11e9-8aa5-a6452b139817.png">
6. In the second terminal, you will see the following:
<img width="1140" alt="Screenshot 2019-07-23 at 10 24 32" src="https://user-images.githubusercontent.com/23398223/61691443-3de78b00-ad34-11e9-84bd-8acbf18a5c14.png">

6. This means the service of updating the calendars channels is running as it's supposed to without having to hit the endpoint manually. For testing purposes, refreshing will happen once a minute but in production, it will happen once a day.

#### Checklist



#### JIRA
[CON-137](https://andela-apprenticeship.atlassian.net/jira/software/projects/CON/boards/15?selectedIssue=CON-137)
